paddle/fluid/framework/tensor.h

```diff
@@ -37,23 +41,51 @@ class Tensor {
 #ifdef PADDLE_WITH_MKLDNN
 
  public:
-  inline mkldnn::memory::format format() const { return format_; }
+  // TODO(jczaja): This is depracted and will be removed
+  inline mkldnn::memory::format format() const {
+    if (layout_ == DataLayout::kMKLDNN) {
+      return static_cast<mkldnn::memory::format>(mem_pd_.desc().data.format);
+    } else {
+      return mkldnn::memory::format::format_undef;
+    }
+  }
+
+  // TODO(jczaja): This is depracted and will be removed
+  inline void set_format(
+      const mkldnn::memory::format fmt,
+      mkldnn::memory::data_type data_type = mkldnn::memory::f32) {
+    mem_pd_ = paddle::platform::create_prim_desc_from_format(
+        paddle::framework::vectorize2int(dims()), fmt, data_type);
+    layout_ = DataLayout::kMKLDNN;
+  }
 
-  inline void set_format(const mkldnn::memory::format format) {
-    format_ = format;
+  inline mkldnn::memory::primitive_desc get_prim_desc() const {
+    return mem_pd_;
+  }
+
+  inline void set_prim_desc(mkldnn::memory::primitive_desc& mem_pd) {
+    // Internally MKL-DNN is just copying (increasing reference counter)
+    // to shared_ptr. So asignment should be quite cheap
+    mem_pd_ = mem_pd;
+    layout_ = DataLayout::kMKLDNN;
+  }
+
+  inline void create_prim_desc_from_dims(
```

Comment:
I think the whole function can be moved to util? Then call set_mkldnn_prim_desc?

Reply:
Ok, I moved whole functions to util and called set_mkldnn_prim_desc